pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -81,11 +90,36 @@ type CertificateManager interface {
 	AddToStatusManager(manager status.StatusManager, namespace string)
 	// KeyPair Returns the CA KeyPairInterface, so it can be rendered in the operator namespace.
 	KeyPair() certificatemanagement.KeyPairInterface
+	// Loads an existing trusted bundle to pass to render.
+	LoadTrustedBundle(context.Context, client.Client, string) (certificatemanagement.TrustedBundleRO, error)
+}
+
+type Option func(cm *certificateManager) error
+
+func AllowCACreation() Option {
+	return func(cm *certificateManager) error {
+		cm.allowCACreation = true
+		return nil
+	}
+}
+
+func WithLogger(log logr.Logger) Option {
+	return func(cm *certificateManager) error {
+		cm.log = log
+		return nil
+	}
+}
+
+func WithTenant(t *operatorv1.Tenant) Option {
+	return func(cm *certificateManager) error {
+		cm.tenant = t
+		return nil
+	}
 }
 
 // Create creates a signer of new certificates and has methods to retrieve existing KeyPairs and Certificates. If a user
 // brings their own secrets, CertificateManager will preserve and return them.
-func Create(cli client.Client, installation *operatorv1.InstallationSpec, clusterDomain, ns string) (CertificateManager, error) {
+func Create(cli client.Client, installation *operatorv1.InstallationSpec, clusterDomain, ns string, opts ...Option) (CertificateManager, error) {
```

Comment:
Do you know where the `clusterDomain` is used? I see it set to an exported variable, but I can't find where it's actually used.

Reply:
Hm, yeah it seems like that variable is never used.

Reply:
@rene-dekker can you remember what the intention for that variable was?

Reply:
I think in my initial implementation the manager was trying to compute the dns names for the leaf certs until I realized that this could not be done.